tests/test_body_spherical_barrier.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 Stéphane Caron
+
+"""Tests for position barrier limit."""
+
+import unittest
+
+import numpy as np
+from robot_descriptions.loaders.pinocchio import load_robot_description
+
+from pink import Configuration
+from pink.barriers import BodySphericalBarrier, NegativeMinimumDistance
+
+
+class TestBodySphericalBarrier(unittest.TestCase):
+    """Testing position barrier limit."""
+
+    def setUp(self):
+        """Set test fixture up."""
+        robot = load_robot_description("yumi_description")
+        model = robot.model
+        self.data = robot.data
+        self.model = model
+        self.robot = robot
+        self.ees = ("yumi_link_7_l", "yumi_link_7_r")
+        self.q0 = np.array(
+            [
+                0.045,
+                -0.155,
+                -0.394,
+                -0.617,
+                -0.939,
+                -0.343,
+                -1.216,
+                0,
+                0,
+                -0.374,
+                -0.249,
+                0.562,
+                -0.520,
+                0.934,
+                -0.337,
+                1.400,
+                0,
+                0,
+            ]
+        )
+        self.configuration = Configuration(self.model, self.data, self.q0)
+
+    def test_negative_distance(self):
+        """Raise an error when the robot body is not found."""
+        with self.assertRaises(NegativeMinimumDistance):
+            BodySphericalBarrier(self.ees, d_min=-1)
+
+    def test_dimension(self):
+        """Check dimensions of configuration limit projection."""
+        self.assertEqual(BodySphericalBarrier(self.ees, d_min=0.2).dim, 1)
+
+    def test_gains(self):
+        """Check gains of configuration limit projection."""
+        # One limit, scalar gain
+        self.assertEqual(
+            BodySphericalBarrier(self.ees, d_min=0.2).gain.shape, (1,)
+        )
+
+    def test_jacobians(self):
+        """Test that shapes of jacobians in all barriers are correct."""
+
+        barrier = BodySphericalBarrier(self.ees, d_min=0.2)
+        J = barrier.compute_jacobian(self.configuration)
+        self.assertEqual(J.ndim, 1)
+        self.assertEqual(J.shape[0], self.robot.nv)
```

Comment:
If my comment above on the sign of `pos2_jac` is correct, the issue was not caught by this test.

Can we add a test on the fixture's `self.configuration` to check the contents of the Jacobian and cover this?

Reply:
We are afraid that we did not get the idea. Do you suggest taking a simple model, precalculating a Jacobian and comparing it with the results of a barrier?

Also, we believe that sign is correct since we are doing:  
```python
np.concatenate( [pos1_world - pos2_world, pos2_world - pos1_world])
```
for `dh_dx`

Reply:
A simple (but shallow) way to test Jacobians is to compute the numerical value of one for a given robot in a given configuration, and have the unit test just check that the Jacobian matrix returned by the function matches the known one.

A more principled way to test Jacobian functions is to compare them against finite difference, as done in [these Task unit tests](https://github.com/stephane-caron/pink/blob/0467084315532d89fb00df7e6b02efc8e079bf0f/tests/test_jacobians.py#L41). You could adapt the same logic to test barriers in a systematic way, but this can be done in a later PR.